src/Clause.ts

```diff
@@ -316,3 +324,17 @@ export default class Clause extends Builder {
   static elements = ['EMU-INTRO', 'EMU-CLAUSE', 'EMU-ANNEX'];
   static linkElements = Clause.elements;
 }
+
+function parsedHeaderToSignature(parsedHeader: ParsedHeader): Signature {
+  return {
+    parameters: parsedHeader.params.map(p => ({
+      name: p.name,
+      type: p.type,
+    })),
+    optionalParameters: parsedHeader.optionalParams.map(p => ({
+      name: p.name,
+      type: p.type,
+    })),
+    return: parsedHeader.returnType,
+  };
+}
```

Comment:
An empty line is forgotten here. BTW, maybe a check should be added into a linting stage?